src/backend/metal/src/command.rs

```diff
@@ -2242,14 +2282,19 @@ impl com::RawCommandBuffer<Backend> for CommandBuffer {
         T: IntoIterator,
         T::Item: Borrow<com::ImageCopy>,
     {
-        let mut inner = self.inner.borrow_mut();
+        let CommandBufferInner {
+            ref mut retained_textures,
+            ref mut sink,
+            ..
+        } = *self.inner.borrow_mut();
+
         let new_src = if src.mtl_format == dst.mtl_format {
-            src.raw.clone()
+            &src.raw
         } else {
             assert_eq!(src.format_desc.bits, dst.format_desc.bits);
             let tex = src.raw.new_texture_view(dst.mtl_format);
-            inner.retained_textures.push(tex.clone());
-            tex
+            retained_textures.push(tex);
+            retained_textures.last().unwrap()
```

Comment:
Seems a bit indirect to pull `tex` back out like this, is there another way?

Reply:
I don't think so - we happen to use a similar patter a bit in WebRender. I think it's just a lack of Entry-like API for vectors.